dnf5/context.cpp

```diff
@@ -321,6 +321,27 @@ void Context::Impl::load_repos(bool load_system) {
 }
 
 void Context::Impl::store_offline(libdnf5::base::Transaction & transaction) {
+    // Test the transaction
```

Comment:
Scenario:
1. a new transaction is tested
2. `get_status()` of the offline transaction status is tested - if there is already an offline transaction in the queue and an `AbortedbyUserError` exception is thrown, the tested transaction is discarded
3. serialize the transaction under test

Can we change the order? Test new transaction only if `get_status()` test passes?
1. `get_status()` of offline transaction status is tested
2. a new transaction is tested
3. serialize the transaction under test

Reply:
That's a good point. Maybe we should do the status test even before downloading transaction packages.

Reply:
I've added a commit with such optimization. Now the offline transaction status is checked before the download phase.